user/user.go

```diff
@@ -186,17 +188,56 @@ func (u *User) apply(rootDir string) error {
 		return errors.Wrap(err)
 	}
 
-	args = []string{
-		"chpasswd",
-		"--root",
-		rootDir,
-		"-e",
+	if u.Password != "" {
+		args = []string{
+			"chpasswd",
+			"--root",
+			rootDir,
+			"-e",
+		}
+
+		pwd := fmt.Sprintf("%s:%s", u.Login, u.Password)
+
+		if err := cmd.PipeRunAndLog(pwd, args...); err != nil {
+			return errors.Wrap(err)
+		}
 	}
 
-	pwd := fmt.Sprintf("%s:%s", u.Login, u.Password)
+	if len(u.SSHKeys) > 0 {
+		if err := writeSSHKey(rootDir, u); err != nil {
+			return err
+		}
+	}
 
-	if err := cmd.PipeRunAndLog(pwd, args...); err != nil {
-		return errors.Wrap(err)
+	return nil
+}
+
+func writeSSHKey(rootDir string, u *User) error {
+	dpath := filepath.Join(rootDir, "home", u.Login, ".ssh")
+	fpath := filepath.Join(dpath, "authorized_keys")
+
+	if err := utils.MkdirAll(dpath, 0755); err != nil {
+		return err
+	}
+
+	f, err := os.OpenFile(fpath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
```

Comment:
These permissions are too permissive
```suggestion
	f, err := os.OpenFile(fpath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
```